### DISCUSSION TITLE
Getting 500 Internal Server Error on gh classroom clone & accepted-assignments

### DISCUSSION THREAD

**Discussion by KayvanShah1** [community/community#67264] [GitHub Education]
### Hi everyone,

The `gh classroom` is authenticated properly yet facing `500 INTERNAL SERVER ERROR` issue.

```bash
Run gh classroom clone student-repos -a $ASSIGNMENT_ID --per-page 100
2023/09/16 06:24:37 HTTP 500: Server Error (https://api.github.com/assignments/480995/accepted_assignments?page=1&per_page=100)
Error: Process completed with exit code 1.
``` 

The API response yet shows
```json
{
  "message": "Requires authentication",
  "documentation_url": "https://docs.github.com/rest/classroom/classroom#list-accepted-assignments-for-an-assignment"
}
```

**Comment by saggu** [comment]
Bumping it up for visibility

**Comment by kate-holdener** [comment]
I'm seeing the same problem.

**Comment by DrErle** [comment]
Still happening @ 1415 EST.

**Comment by jeffbcit** [comment]
I'm seeing the same thing. Trying to clone all my student's assignments and getting the "HTTP 500: Server Error"

**Comment by kate-holdener** [comment]
I was able to work around this (for now) using the classroom assistant: https://github.com/education/classroom-assistant/releases
That's the tool I used last school year, which is now deprecated (but is the only way I can efficiently get all the repos cloned).

**Reply by DrErle** [reply]
I was able to re-download and install Classroom Assistant and use that successfully. 

Does anyone from GitHub support read these posts? I was expecting a post along the lines of, "We're aware of the issue and are actively working it. Will post regularly with updates until resolved."

**Comment by tsmeyer** [comment]
also having the same issue

**Comment by nondescryptid** [comment]
Same here

**Comment by fhnw-webec** [comment]
Same here

**Reply by lnocera** [reply]
Same here!

**Comment by Kostikyan** [comment]
I'm seeing the same problem...

**Comment by gabrielproulx3** [comment]
Same problem for me and other teachers when trying to use gh classroom

**Comment by rkumar-teaching** [comment]
same here :(

**Comment by sprenks18** [comment]
Same here.  This worked for me two days ago.  I'm not sure what's wrong/different.

**Comment by kate-holdener** [comment]
@RyanHecht - can GitHub Classroom team help with this issue?

**Comment by AlexDmr** [comment]
Similar problem :
```typescript
octokit.request(`GET /assignments/${id}/accepted_assignments`, {
    assignment_id: `${id}`,
    headers: {
      'X-GitHub-Api-Version': '2022-11-28'
    }
  })
```

result in a HTTP error 500 
```json
{
  "message": "Server Error",
  "documentation_url": "https://docs.github.com/rest/classroom/classroom#list-accepted-assignments-for-an-assignment"
}
```

**Comment by RyanHecht** [comment]
Hey folks! There are two parallel issues going on here, let me try and provide some clarity :)

Yesterday we deployed a bad webserver configuration that was making nearly _every_ API call return 500 errors. This change has been reverted, so the issues @AlexDmr @kate-holdener @sprenks18 @rkumar-teaching @gabrielproulx3  @fhnw-webec @nondescryptid were having yesterday should be resolved!

Separately, we're noticing 500 errors when attempting to clone student repos using sufficiently large arguments to `--per-page` -- this is because of some scaling issues with the API call. The team is presently working on a longer term fix for this (including more friendly options to download _all_ student submissions at once), but for now the workaround is to use smaller page sizes

We have [an issue](https://github.com/github/gh-classroom/issues/38) in the CLI extension's repository where we've been discussing this!

**Reply by AlexDmr** [reply]
Hello @RyanHecht ,
thanks for the help. 
Unfortunatly the problem does remain for me (same requests, same errors 500.
Authentification is done, I can get classrooms and assignements but not accepted assignements...

**Reply by AlexDmr** [reply]
Oups not exactly true, actually some request are OK (code 200, plus the data), and some are 500.
It may be because I send too much requests ?

**Reply by profedbrown** [reply]
Still a problem.  

2023/09/23 10:54:28 HTTP 500: Server Error (https://api.github.com/assignments/485736/accepted_assignments?page=1&per_page=30)


Could not find a way to adjust or can I set the page length on the classroom web page?

**Reply by nondescryptid** [reply]
Hi Ryan, thanks for the update :) I'm able to clone my students' repos now (~30 students).

**Reply by RyanHecht** [reply]
@profedbrown You can use the `--per-page` and `--page` flags to use the pagination feature!

**Reply by sprenks18** [reply]
I was able to do what I wanted on Friday with 

   gh classroom clone student-repos 
   gh classroom clone student-repos --page 2

Today, on the first request, I get an error 
2023/09/25 14:54:38 HTTP 500: Server Error (https://api.github.com/assignments/493095/accepted_assignments?page=1&per_page=30)

but then I am able to successfully clone the second page (which is about 8 students).

**Reply by girodg** [reply]
@RyanHecht  Doesn't work for me. It won't list the assignments regardless of page size.
I get this when put in page size of 10.
2023/09/28 13:29:49 HTTP 500: Server Error (https://api.github.com/classrooms/26315/assignments?page=1&per_page=100)

**Comment by SumanthSShet** [comment]
The error message "Requires authentication" suggests that the gh classroom tool is not properly authenticated to GitHub. This could be due to a number of factors, such as:

The gh classroom tool is not using the correct OAuth token.
The OAuth token has expired.
The OAuth token does not have the necessary permissions.
To troubleshoot this error, you can try the following:

Make sure that you are using the correct OAuth token. You can find the OAuth token in the gh classroom tool's settings.
Make sure that the OAuth token has not expired. You can check the expiration date of the OAuth token in the gh classroom tool's settings.
Make sure that the OAuth token has the necessary permissions. The gh classroom tool needs the following permissions:
read:user
write:user
read:org
If you are still having trouble, you can contact GitHub support for assistance.

Here are some additional tips for troubleshooting gh classroom errors:

Check the gh classroom documentation for more information about the error.
Try running gh classroom with the --verbose option to get more detailed error messages.
Try running gh classroom from a different computer or network.
I hope this helps!

**Comment by girodg** [comment]
Mine won't even get passed the classroom selection to select an assignment no matter what I choose for per-page nor page.
This is the error I get. 
(https://api.github.com/classrooms/26315/assignments?page=1&**per_page=100**)

NOTE: it always lists per_page=100 regardless of what I enter for per-page.

**Comment by mbaumanndev** [comment]
On my side I have this issue today right now with my assignments, it's painful that the UI forces us to download each repos one by one or use the CLI to get all the assignments, since the CLI doesn't work (I hve updatage both gh cli and classroom extension)

**Comment by yuhengy** [comment]
Same issue here since yesterday.

**Comment by yuhengy** [comment]
The problem is still here. Let me provide more details:

After students click the link to accept the assignment. The page below will pop up.

<img width="300" alt="image" src="https://github.com/community/community/assets/40620196/d9b619cc-8557-484f-af45-056473afcadf">

Students in our class are accepting same few assignments in last few weeks. Some students accepted earlier and some later. Before last week, everything is working properly, there is not such error page. However, just a few days ago, many students start to see this error. So I think I have everything configured correctly but the GitHub Classroom itself has some problem.

However, even this error page pops up, the repos for students are indeed created. Students (and instructors) cannot access the repo by clicking the buttons (shown below) in GitHub Classroom. After clicking it, another error page will pop up (shown below). But they can access the repos through the repo links, such as https://github.com/{OrganizationName}/{AssignmentName}-{UserName}.

<img width="1000" alt="Screenshot 2024-02-14 at 04 51 41" src="https://github.com/community/community/assets/40620196/c704830a-df67-4b11-8440-122c46e2b845">

<img width="300" alt="Screenshot 2024-02-14 at 04 51 48" src="https://github.com/community/community/assets/40620196/b20d8a12-d947-4984-abeb-46e5d1864d26">

Have anyone experience the same problem recently? Even though I can alway ask students to use the repo link to access the repo, it is still annoying that we cannot access the repo in the GitHub Classroom Webpage. Is there anything can be done to avoid the error page and improve students' using experience?

**Reply by Delonvdv** [reply]
I have been getting the same errors as mentioned above since last week

**Comment by joaquinvanschoren** [comment]
We have the same problem. Our assignment has been running for a week and we haven't changed anything. Today, suddenly some students who recently started encountered this issue. Will this be resolved soon?
